src/Language/PureScript/TypeChecker/Types.hs

```diff
@@ -74,7 +74,7 @@ typesOf ::
   [(Ident, Expr)] ->
   m [(Ident, (Expr, Type))]
 typesOf bindingGroupType moduleName vals = do
-  ((tys, subst), warnings) <- reflectErrors . censor (const mempty) . reifyErrors . listen $ liftUnify $ do
+  ((tys, subst), warnings) <- reflectErrors . censor (const mempty) . reifyErrors . listen $ withEmptySubstitution $ do
```

Comment:
I'm not sure how to tell whether it's behaving correctly or not actually. What is the purpose of reify and then reflecting? The main thing I was concerned about here was "replacing" the warnings with `censor` and then `tell` just after.